### ISSUE TITLE
`search` needs more `log.debug()` statements

### ISSUE THREAD

**Issue by barentsen** [KeplerGO/lightkurve#539]
[Over on Twitter](https://twitter.com/kellecruz/status/1144310082613723137) it is rightly pointed out that it can be hard to diagnose why `search_targetpixelfile` hangs.  I propose we add a number of useful `log.debug()` statements to the `search` module such that one can diagnose issues more easily, e.g. using:
```python
import lightkurve as lk
lk.log.setLevel("DEBUG")
lk.search_targetpixelfile(...)
``` 

Everything is already in place to support the example above, we just need to add `log.debug()` statements in useful places. Optionally, we can consider supporting debug mode using:

```python
lk.search_targetpixelfile(..., verbose=True)
```

**Comment by barentsen** [comment]
(we need to look into an optional download progress bar as well)

**Comment by barentsen** [comment]
We merged #547 today, which enables users to see status messages during search and download operations by calling `lightkurve.log.setLevel("DEBUG")` prior to executing search or download functions.

This will be part of the Lightkurve v1.1 release.

**Comment by barentsen** [comment]
Example of the new functionality:
![Screenshot from 2019-07-18 12-34-43](https://user-images.githubusercontent.com/817669/61486503-76641f00-a958-11e9-9d6c-dabf2b77598c.png)
